waiter/integration/waiter/authentication_test.clj

```diff
@@ -17,12 +16,14 @@
                                                       :run-as-user (retrieve-username)
                                                       :token token)
                                                     :authentication))]
-        (assert-response-status response 200)
-        (let [{:keys [service-id body]} (make-request-with-debug-info {:x-waiter-token token} #(make-kitchen-request waiter-url % :path "/request-info"))
-              body-json (json/read-str (str body))]
-          (with-service-cleanup
-            service-id
-            (is (= (retrieve-username) (get-in body-json ["headers" "x-waiter-auth-principal"])))))))))
+        (try
+          (assert-response-status response 200)
+          (let [{:keys [service-id body]} (make-request-with-debug-info {:x-waiter-token token} #(make-kitchen-request waiter-url % :path "/request-info"))
+                body-json (json/read-str (str body))]
+            (with-service-cleanup
+              service-id
+              (is (= (retrieve-username) (get-in body-json ["headers" "x-waiter-auth-principal"])))))
+          (finally (delete-token-and-assert waiter-url token)))))))
```

Comment:
Nitpick: Can we get the expression following the finally in a new line?

Reply:
done